### PR TITLE
Remove inefficient loop in TopP logits processor

### PR DESCRIPTION
**Pull request by tohtana**
The loop in Top P processor is inefficient when it runs on multiple requests.
This PR replaces the loop with a batch API with indexing.

### CODE REVIEW

**Comment by tohtana** [comment]
Closed because an inappropriate commit is included
